jax/experimental/array.py

```diff
@@ -98,6 +98,13 @@ def _reconstruct_array(fun, args, arr_state, aval_state):
 class Array:
   # TODO(yashkatariya): Add __slots__ here.
 
+  aval: core.ShapedArray
```

Comment:
Given that you are adding this, mind adding __slots__ too?

Reply:
Sure, I'll give that a try here.

Reply:
Hm, turns out we can't use `__slots__` because the `pxla.maybe_cached_property` relies on the class having a `__dict__`. I suppose we could add `"__dict__"` to `__slots__`, but that would kind of defeat the purpose 😀

I'll revert the `__slots__` change for now, and leave the TODO in place.

Reply:
Sounds good! Thanks for trying!